### PR TITLE
[WIP] Add "hidden" property attribute

### PR DESCRIPTION
**Pull request by gramian**
## What does this PR do?

This new boolean property attribute called `hidden` excludes properties from a projection containing a `*` (star).

## Motivation

[ZetaSQL](https://github.com/google/zetasql/blob/master/docs/data-definition-language.md?plain=1#L204)

## Related issues

https://github.com/ArcadeData/arcadedb/discussions/1439

## Additional Notes

This is the interesting change set: https://github.com/ArcadeData/arcadedb/pull/1503/files#diff-1ebfff8995bc54eab127a05a7f09d799b215972c6aa872635f37d86e824fb0b3

* Basically now the any projection with a `*` (`isAll`) gets parsed property by property not by copy any more.
* Particularly I don't understand why the argument record is copied into the result as whole (see https://github.com/ArcadeData/arcadedb/pull/1503/files#diff-1ebfff8995bc54eab127a05a7f09d799b215972c6aa872635f37d86e824fb0b3R108 )?
* `SELECT * FROM x` works, but `SELECT FROM x` does not.

## Checklist

- [ ] I have run the build using `mvn clean package` command
- [ ] My unit tests cover both failure and success scenarios

### CODE REVIEW

File: engine/src/main/java/com/arcadedb/schema/DocumentType.java [hunk 1]
```diff
@@ -87,6 +87,8 @@ default Property createProperty(String propName, JSONObject prop) {
       p.setMandatory(prop.getBoolean("mandatory"));
     if (prop.has("notNull"))
       p.setNotNull(prop.getBoolean("notNull"));
+    if (prop.has("hidden"))
+      p.setNotNull(prop.getBoolean("hidden"));
```
Comment (lvca) [inline engine/src/main/java/com/arcadedb/schema/DocumentType.java]: This should be `.setHidden()`, right?

Comment (gramian) [reply engine/src/main/java/com/arcadedb/schema/DocumentType.java]: 🤦 right, a classic copy/paste error. Fixed but still not working though...
Thanks for checking

**Review by lvca — APPROVED** [review]

**Comment by lvca** [comment]
I reviewed this PR and it's good for me. What else is pending to close it?

**Comment by gramian** [comment]
* It is not working in all situations as expected
* There are no tests yet

I look into this on tomorrow and try to finish it.

**Pull request closed without merging**